WebDriverAgentLib/Categories/XCElementSnapshot+FBHelpers.m

```diff
@@ -76,6 +76,12 @@ - (DDXMLElement *)XMLElementFromElement:(XCElementSnapshot *)snapshot indexPath:
   if (snapshot.wdLabel) {
     [xmlElement addAttribute:[DDXMLNode attributeWithName:@"label" stringValue:snapshot.wdLabel]];
   }
+  if (snapshot.wdVisible) {
+      [xmlElement addAttribute:[DDXMLNode attributeWithName:@"isVisible" stringValue:snapshot.wdVisible ? @"1" : @"0"]];
```

Comment:
You will never add `isVisible` and `isEnabled`, when they are false. This means that any xpath that would search for `0` value will fail right?